core/src/main/java/hudson/model/View.java

```diff
@@ -1006,6 +1009,36 @@ public synchronized void doDoDelete(StaplerRequest req, StaplerResponse rsp) thr
     public abstract Item doCreateItem( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException;
 
     /**
+     * Makes sure that the given name is good as a job name.
+     * For use from {@code newJob}.
+     */
+    @Restricted(DoNotUse.class) // called from newJob view
+    public FormValidation doCheckJobName(@QueryParameter String value) {
+        // this method can be used to check if a file exists anywhere in the file system,
+        // so it should be protected.
+        getOwner().checkPermission(Item.CREATE);
+
+        if (Util.fixEmpty(value) == null) {
+            return FormValidation.ok();
+        }
+
+        try {
+            Jenkins.checkGoodName(value);
```

Comment:
Exactly, I'm thinking about a client-side validation like: `required`, `illegal-characters`, etc... Really, I'm trying to avoid that the users reach this no pretty view:

![jenkins-validation](https://cloud.githubusercontent.com/assets/1021745/14117801/93386614-f5e6-11e5-8940-d79e171572c2.png)